### ISSUE TITLE
write_dta: numeric labelleds not written correctly

### ISSUE THREAD

**Issue by diogocp** [hadley/haven#144]
I caught this in #140 but have not yet discovered the cause of the problem.
```
> num <- labelled(c(1, 2), c(a = 1, b = 3))
> num
<Labelled>
[1] 1 2

Labels:
 value label is_na
     1     a FALSE
     3     b FALSE
> roundtrip_var(num)
<Labelled>
[1] 1 2

Labels:
 value label is_na
     0     a FALSE
     0     b FALSE
```

I opened the file in Stata and verified that the labels are not written correctly:
```
. label list
x..i..:
           0 a
           0 b
```

It might be a bug in ReadStat, since the [label-writing code](https://github.com/hadley/haven/blob/master/src/DfWriter.cpp#L165) in DfWriter seems to be basically the same as the code for integer labelleds, which do work fine.

**Comment by evanmiller** [comment]
Stata only supports value labels for integers:

http://www.stata.com/help.cgi?dta#value_labels

You could try creating the label set for integer values (`readstat_label_int32_value`) but I am not sure what the Stata behavior is when an integer label set is applied to a double-precision column.

**Comment by diogocp** [comment]
You're right. I thought I had tried to label a double in Stata, but apparently I did not.
```
. label define test 3.14 "pi"
may not label 3.14
```
Applying an integer label to a double variable does work, but is probably not very useful.
```
. list, clean

          x  
  1.      1  
  2.   3.14  

. label define test 1 "one"
. label values x test
. list, clean

          x  
  1.    one  
  2.   3.14 
```

The easiest way to fix this would be to warn and discard labels when writing numerics. Writing only those labels with integer values could be very surprising for users.
